boards: actinius_icarus_*: fix regressions, add support for Icarus hardware v2

alextsam opened the pull request:
This PR includes the following:

- Fixes for regressions in #43817
- Fixes for pre-existing issues
- Support for Icarus IoT Board hardware version v2

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam committed:
boards: actinius_icarus_*: only include `board.c` in non-secure targets

Refrain from including `board.c` in secure targets as a workaround for
SPM having the GPIO driver disabled.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam committed:
boards: actinius_icarus_*: board.c: fix pin initialization

Revert back to using `GPIO_OUTPUT_*` instead of `GPIO_OUTPUT_INIT_*`
as the later does not including enabling the GPIO output.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam committed:
boards: actinius_icarus_*: fix `sim-select` binding docs about enum use

Change the documentation of the `sim-select` DT binding to explain
the proper use of the configuration property using enum values.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam committed:
boards: actinius_icarus_bee: fix incorrect `sim-select` gpio

Fix the pin number of the `sim-select` gpio to the correct one
for the Icarus Bee.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam committed:
boards: actinius_icarus_som: fix incorrect `sim-select` gpio

Fix the pin number of the `sim-select` gpio to the correct one
for the Icarus SoM.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam committed:
boards: actinius_icarus_som: fix accelerometer INT pins

Update the DT with the correct order of the INT1 and INT2 pins
of the accelerometer.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam committed:
boards: actinius_icarus_bee: fix accelerometer INT pins

Update the DT with the correct order of the INT1 and INT2 pins
of the accelerometer.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam committed:
boards: actinius_icarus_*: align def RAM plan with that of the nRF9160DK

Align the default RAM allocation to be according to that of the
nRF9160DK by reducing the reserved SRAM for the modem lib.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam committed:
boards: actinius_icarus: change bootloader/mcuboot LED to blue

Change the DT properties related to the bootloader/mcuboot
to set the LED to blue.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam committed:
boards: actinius_icarus_bee: change bootloader/mcuboot LED to blue

Change the DT properties related to the bootloader/mcuboot
to set the LED to blue.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam commented:
cc @gmarull 
cc @carlescufi

galak commented on boards/arm/actinius_icarus_som/dts/bindings/sim-select.yaml:
```diff
@@ -4,8 +4,8 @@
+  external nano-SIM. The SIM can be selected by changing the `sim` property
+  of the `sim_select` dt node to `esim` or `external`.
 
 compatible: "sim-select"
```
Since you are update a number of things, the `compatible` should really be something like: `actinius,sim-select`

galak commented on boards/arm/actinius_icarus/dts/bindings/charger-enable.yaml:
```diff
@@ -0,0 +1,26 @@
+  the `charger` property in the `charger_enable` dt node to `auto`
+  or `disabled`.
+
+compatible: "charger-enable"
```
actinius,charger-enable

galak reviewed (dismissed):
Update dts compatibles.

alextsam commented:
> Update dts compatibles.

Done!

gmarull commented on boards/arm/actinius_icarus/dts/bindings/sim-select.yaml:
```diff
@@ -4,8 +4,8 @@
   LTE-M/NB-IoT communication. Either using the on-board eSIM or using an
-  external nano-SIM. The SIM can be selected using the SIM select pin given by
-  the sim-gpios property by setting it to 0 (eSIM) or to 1 (external nano-SIM).
+  external nano-SIM. The SIM can be selected by changing the `sim` property
+  of the `sim_select` dt node to `esim` or `external`.
 
```
I see, it's yaml not rst, my fault!

gmarull commented on boards/arm/actinius_icarus/actinius_icarus_common.dts:
```diff
@@ -86,7 +86,7 @@
 
 	sim_select: sim-select {
-		compatible = "sim-select";
+		compatible = "actinius,sim-select";
```
I was precisely told not to do this: https://github.com/zephyrproject-rtos/zephyr/pull/43817#discussion_r831620227 cc: @mbolivar-nordic

gmarull replied:
@mbolivar-nordic ping

alextsam replied:
@mbolivar-nordic @galak I can change it back, but can we please make sure there is a consensus first?

mbolivar-nordic replied:
Sounds good -- @galak, you OK with removing the invalid vendor prefix?

alextsam replied:
@galak ping

gmarull replied:
@galak ping

alextsam replied:
@galak @gmarull @mbolivar-nordic could you please let me know whether it should be with or without the vendor prefix so that we get these fixes in as soon as possible? The board files are broken as they are in zephyr right now.

gmarull commented on boards/arm/actinius_icarus/actinius_icarus_common_2_0_0.dtsi:
```diff
@@ -0,0 +1,57 @@
+ */
+
+&pinctrl {
+	uart0_default_v2: uart0_default_v2 {
```
why "v2"? that means original common is no longer common, so you should likely remove `uart0_default` from common?

alextsam replied:
`v2` is for "v2 hardware" (there are 2 revisions of the Icarus in the definitions now). `common` is for "common between secure and not secure". I hope this clarifies things

alextsam replied:
uart0 of the v2 revision is "overlay-ed" to use this v2 pinctrl, see ref to line 28

gmarull replied:
so there's something wrong: v1 uart is not common and so it should not be in the common file. just have a true common file, a v1 dts and a v2 dts.

alextsam replied:
I am not sure what you see as wrong here; there is a "base" DTS which belongs to the v1.4 rev and an overlay that adjusts for v2. This is exactly the same way it is done in other boards. See for instance nrf9160dk_nrf9160 v14 spi3.
And to re-iterate, "_common" has the meaning of "common between secure and non-secure" in this context.

gmarull replied:
Please check https://github.com/teslabs/zephyr/commit/541d569be322bc3e97d38bfedd89f5424c478f8b

gmarull commented on boards/arm/actinius_icarus/actinius_icarus_common_2_0_0.dtsi:
```diff
@@ -0,0 +1,57 @@
+	};
+};
+
+/ {
+	charger_enable: charger-enable {
+		compatible = "actinius,charger-enable";
+		gpios = <&gpio0 7 GPIO_ACTIVE_LOW>;
+		charger = "auto";
+	};
+};
```
can't regulator be used? grep for `regulator-fixed` in some boards.

alextsam replied:
I guess it technically could, but I don't think it's fitting, because this is not a device that is on or off; the charger can be disabled, otherwise it has it's own logic on when it is on (i.e. why the `auto` enum value).

gmarull commented on boards/arm/actinius_icarus/dts/bindings/actinius,charger-enable.yaml:
```diff
@@ -0,0 +1,26 @@
+description: |
+  The Icarus IoT Board v2 provides the user with an option to disable the
+  charger or to leave it on auto. The option can be changed by changing
+  the `charger` property in the `charger_enable` dt node to `auto`
```
compatibles should not reference hardcoded node names?

alextsam replied:
Updated with generic wording, thanks!

carlescufi commented:
@gmarull please revisit

alextsam commented on boards/arm/actinius_icarus/actinius_icarus_common_2_0_0.dtsi:
```diff
@@ -0,0 +1,57 @@
+	status = "okay";
+
+	current-speed = <115200>;
+	pinctrl-0 = <&uart0_default_v2>;
```
@gmarull uart0 pins change here

alextsam commented on boards/arm/actinius_icarus/board.c:
```diff
@@ -11,27 +11,68 @@
+
+	int result;
+
+	result = set_sim_select_pin();
+	if (result < 0) {
+		LOG_ERR("Failed to set the SIM Select Pin (error: %d)", result);
+		/* do not return so that the rest of the init process is attempted */
+	}
+
+	result = set_charger_enable_pin();
+	if (result < 0) {
+		LOG_ERR("Failed to set the Charger Enable Pin (error: %d)", result);
+		/* do not return so that the rest of the init process is attempted */
+	}
```
These errors are logged to the active logging backend (e.g. UART or flash)

alextsam commented on boards/arm/actinius_icarus/board.c:
```diff
@@ -11,27 +11,68 @@
+		/* do not return so that the rest of the init process is attempted */
+	}
+
+	return result;
```
`result` can be set to an error code from the `set_sim_select_pin()` or `set_charger_enable_pin()`

galak commented on boards/arm/actinius_icarus/dts/bindings/actinius,charger-enable.yaml:
```diff
@@ -0,0 +1,26 @@
+      required: true
+      description: Pin used to enable/disable the charger
+
+    charger:
```
If there are only 2 states like this, wouldn't a boolean make more sense?

alextsam replied:
Technically it can be a bool; would that be preferable? In my opinion the enums were more fitting because the charger can be disabled, otherwise it has it's own logic about when it is on (i.e. auto). Let me know what you think, thanks

galak commented:
Still one question on `charger` property and if it makes sense for it to be a bool instead

gmarull committed:
boards: arm: actinius_icarus: split non-common pinctrl entries

actinius_icarus_common-pinctrl.dtsi contained entries for uart0 that are
not common for all revisions.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

alextsam commented:
@mbolivar-nordic @gmarull can you please revisit this? The docs seem to support @galak's suggestion, and the nrf9160dk bindings are following the same as well. Why would this be an "invalid vendor prefix"? "actinius" is being added to the vendors file in this PR.

I have rebased to the latest main to resolve conflicts from updates that happened in the meantime

alextsam commented:
@gmarull @mbolivar-nordic ping

galak commented:
> @gmarull @mbolivar-nordic ping

Sorry for the delay.  After some discussion between @mbolivar-nordic and myself, we determined it should be `actinius-sim-select`.  This matches the precedence we have for arduino, in which we use `arduino-header-r3`

alextsam committed:
boards: actinius_icarus_bee: rename sim-select to actinius-sim-select

This renames the `sim-select` binding to `actinius-sim-select`.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam committed:
boards: actinius_icarus_som: rename sim-select to actinius-sim-select

This renames the `sim-select` binding to `actinius-sim-select`.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

galak commented on boards/arm/actinius_icarus/actinius_icarus_common_2_0_0.dtsi:
```diff
@@ -0,0 +1,52 @@
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <80000000>;
+		label = "W25Q64";
```
remove `label`

galak reviewed (changes requested):
remove use of `label` in one dts/overlay file.

alextsam committed:
boards: actinius_icarus: add support for hardware version v2.0

Add support for the new hardware version of the Icarus IoT Board.

Signed-off-by: Alex Tsamakos <alex@actinius.com>

alextsam commented:
> remove use of `label` in one dts/overlay file.

Thank you @galak. I added a commit for this. Should I squash it with d119eddd398f0793c4aa05b0f5c24e5a5b5d163b?

galak commented:
> Thank you @galak. I added a commit for this. Should I squash it with [d119edd](https://github.com/zephyrproject-rtos/zephyr/commit/d119eddd398f0793c4aa05b0f5c24e5a5b5d163b)?

I would squash it :).

alextsam commented:
> > Thank you @galak. I added a commit for this. Should I squash it with [d119edd](https://github.com/zephyrproject-rtos/zephyr/commit/d119eddd398f0793c4aa05b0f5c24e5a5b5d163b)?
> 
> I would squash it :).

Done!

alextsam commented:
@gmarull do you think this will make it into nrf-zephyr in time for the upcoming release? I see that an upmerge is ongoing

gmarull commented:
> @gmarull do you think this will make it into nrf-zephyr in time for the upcoming release? I see that an upmerge is ongoing

Upmerge sha is now fixed, but we could consider adding these patches as well cc: @nordicjm @de-nordic
